### DISCUSSION TITLE
Probleme mit R503-S

### DISCUSSION THREAD

**Discussion by Laudi111** [frickelzeugs/FingerprintDoorbell#57] [Q&A]
Hallo zusammen,
ich versuche seit ein paar Tagen den FingerprintDoorbell Code mit dem Sensor [R503-S](https://www.ebay.de/itm/204016129690) zum Laufen zu bekommen. 

Ich nutze ein ESP32 D1 mini, habe aber auch Tests mit einem ESP8266 und Arduino Uno gemacht.
Grundsätzlich funktioniert der Sensor mit der Adafruit Library und auch mit allen 3 Controllern, allerdings gibt es mit den ESPs Probleme, was sich schon beim ersten auslesen der Daten über getParameters() zeigt.
Dort werden teilweise eine falsche Baudrate oder ID angezeigt. Damit funktioniert in der Folge dann auch der restliche Code nicht mehr.

```
> 13:00:42.686 -> Reading sensor parameters
> 13:00:42.686 -> Status: 0xA
> 13:00:42.686 -> Sys ID: 0x5
> 13:00:42.686 -> Capacity: 0
> 13:00:42.686 -> Security level: 96
> 13:00:42.686 -> Device address: 1EFB3F00
> 13:00:42.686 -> Packet len: 32
> 13:00:42.686 -> Baud rate: 19200
> 13:00:42.686 -> Sensor contains 4 templates
> 13:00:42.686 -> [E][Preferences.cpp:49] begin(): nvs_open failed: NOT_FOUND
> 13:00:42.686 -> 0 fingers loaded from preferences.
> 13:00:42.686 -> [2022-12-20 12:00:42 GMT]: Warning: Fingerprint count mismatch! 4 fingerprints stored on sensor, but we are aware of 0 fingerprints.
> 13:00:42.724 -> [2022-12-20 12:00:42 GMT]: Pairing successful.
```

Das Auslesen der falschen Daten tritt aber scheinbar nur auf, wenn vorher schon mal ein Befehl Richtung R503-S ging, z.B. `finger.LEDcontrol(FINGERPRINT_LED_FLASHING, 25, FINGERPRINT_LED_BLUE, 0); // sensor connected signal`

Wenn ich die kompletten finger.LEDcontrol Zeilen aus dem Code auskommentiere, dann läuft auch der R503-S mit dem ESP32mini und dem FingerprintDoorbell Code, allerdings dann leider komplett ohne LED Feedback.
Wenn ich die LEDcontrol aktiv habe, funktioniert auch keine Kommunikation zum Sensor richtig, es gibt nur Fehlermeldungen und es lassen sich auch keine Fingerabdrücke anlernen oder löschen. Die LEDs funktionieren aber komischerweise.

Ich habe mir auch angeschaut, ob es mit versch. Parametern in der LEDcontrol Funktion zusammenhängt, habe aber leider keinen Zusammenhang ermitteln können, da die falschen getParameters() mal aufgetreten sind und mal nicht.
Ich habe das Gefühl, dass Return Parameter aus der LEDcontrol vermutlich falsch in den nächsten abfragen interpretiert werden, habe aber leider keine Ahnung wie man da weiter in die Analyse gehen könnte.

Hat jemand von euch schon mal versucht einen R503-S in Betrieb zu nehmen? Hier ein Vergleich von R503 und R503-S:

> R503 | R503-5V | R503S
> -- | -- | --
> Netzteil | DC 3,3 V | DC 5V | DC 3,3 V
> Sensing Array | 192*192 pixel | 192*192 pixel | 160*160 pixel
> FingerprintKapazität | 200 | 200 | 150
> Effektive TouchBereich | Durchmesser 15,5mm | Durchmesser 15,5mm | Durchmesser 13,6mm
> Stecker | MX 1,0mm 6pin | MX 1,25mm 6pin | MX 1,0mm 6pin
> 

Das Datenblatt gibts auch hier: https://www.dropbox.com/sh/orprmb3bgb6lqb6/AACpiIXOF91R7-RQ9OkD4JXha?spm=a2g0o.detail.1000023.1.ebbb5fe5M1mr3W&dl=0

Außerdem habe ich noch Resets die der ESP32 durchführt, meistens direkt 1x nach "Started normal operating mode".  Die Resets treten auch mit mehreren verschiedenen ESP32 auf, so dass ich aktuell nicht von einem Problem am ESP32 ausgehe.

```
Adafruit finger detect test
Found fingerprint sensor!
Reading sensor parameters
Status: 0x4
Sys ID: 0x0
Capacity: 150
Security level: 3
Device address: FFFFFFFF
Packet len: 128
Baud rate: 57600
Sensor contains 2 templates
loadFingerListFromPrefs
2 fingers loaded from preferences.
Started normal operating mode

Brownout detector was triggered

ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0018,len:4
load:0x3fff001c,len:1216
ho 0 tail 12 room 4
load:0x40078000,len:10944
load:0x40080400,len:6360
entry 0x400806b4
```

Manchmal kommt auch dieser Reset:
```
[2022-12-20 19:43:36 UTC]: Match Found: 10 - @empty with confidence of 0
[2022-12-20 19:43:37 UTC]: Security issue! Match was not sent by MQTT because of invalid sensor pairing! This could potentially be an attack! If the sensor is new or has been replaced by you do a (re)pairing in settings page.
Unknown error
[2022-12-20 19:43:40 UTC]: ScanResult Error (Code 9)
Guru Meditation Error: Core  1 panic'ed (LoadStoreError). Exception was unhandled.
Core 1 register dump:
PC      : 0x40189ff9  PS      : 0x00060530  A0      : 0x800ebe75  A1      : 0x3ffb1c40  
A2      : 0x4000aa04  A3      : 0x3ffb1c12  A4      : 0x000003e8  A5      : 0x0000000b  
A6      : 0x00000000  A7      : 0x00000008  A8      : 0x3ff58060  A9      : 0x00000003  
A10     : 0x0000000c  A11     : 0x00000043  A12     : 0x00000005  A13     : 0x3ffb1c20  
A14     : 0x00000000  A15     : 0xef010003  SAR     : 0x00000008  EXCCAUSE: 0x00000003  
EXCVADDR: 0x4000aa0a  LBEG    : 0x4000c28c  LEND    : 0x4000c296  LCOUNT  : 0x00000000  

ELF file SHA256: 0000000000000000

Backtrace: 0x40189ff9:0x3ffb1c40 0x400ebe72:0x3ffb1c60 0x400d25e5:0x3ffb1c80 0x400d623a:0x3ffb1ce0 0x400d74e9:0x3ffb1f30 0x400ed939:0x3ffb1fb0 0x40089a7e:0x3ffb1fd0

Rebooting...
``` 

Über Hilfe von euch wäre ich sehr dankbar.

Gruß
Sebastian

**Comment by TheMaskedDeveloper** [comment]
Moin, zu dem Brownout - Resets: Wenn es nicht den den ESP32s liegt, dann versuch mal ein anderes USB-Kabel, USB-Port oder Spannungsquelle:

It’s often related to one of the following issues:

Poor quality USB cable;
USB cable is too long;
Board with some defect (bad solder joints);
Bad computer USB port;
Or not enough power provided by the computer USB port.

Solution: try a different shorter USB cable (with data wires), try a different computer USB port or use a USB hub with an external power supply.

**Comment by ChrisLSaar** [comment]
Hallo,

bei mir exakt die gleichen Probleme mit dem R503-S.
Hab den Sensor aus versehen bestellt weil bei AliExpress nicht richtig gelesen.

Hab es trotzdem versucht und auch mal das Debugging innerhalb er Arduino Fingerprint Lib aktiviert.
Jetzt sieht man schön die Antworten auf die Anfragen auf der UART Schnittstelle.

Für mich sieht das nach einem internen Firmewarethema in dem Sensor aus.
Sobald man mal das LED angesteuert hatte sind die Antworttelegramme auf weitere Anfragen alle falsch.
Ohne Datenlängenangabe und auch mit abgeschnittenen Daten.
Diese werden dann an spätere Daten zum Teil angehängt.

Fragt man zu häufig an kommt es zu dem SW_CPU_RESET im ESP.

Ich forsche nochmals weiter, verspreche mir aber nicht den großen Durchbruch.

Christian
